GRDB/Migration/DatabaseMigrator.swift

```diff
@@ -213,7 +213,17 @@ public struct DatabaseMigrator {
     
     /// Returns the set of applied migration identifiers.
     public func appliedMigrations(in reader: DatabaseReader) throws -> Set<String> {
-        return try reader.read { try appliedIdentifiers($0) }
+        return try reader.read { db in
+            guard let doesMigrationsTableExist = try Bool.fetchOne(db, sql: "SELECT EXISTS (SELECT 1 FROM sqlite_master WHERE type='table' AND name='grdb_migrations')") else {
```

Comment:
I can't help thinking that `appliedMigrations(in:)` is fixed, but the private `appliedIdentifiers` method is still sensitive to the lack of the grdb_migrations tables, for no good reason.

Maybe a better fix would be to fix the root `appliedIdentifiers` method itself.

```swift
public func appliedMigrations(in reader: DatabaseReader) throws -> Set<String> {
    return try reader.read(appliedIdentifiers)
}

private func appliedIdentifiers(_ db: Database) throws -> Set<String> {
    let tableExist = try Bool.fetchOne(db, sql: """
        SELECT EXISTS (SELECT 1 FROM sqlite_master WHERE type='table' AND name='grdb_migrations')
        """)!
    guard tableExist else {
        return []
    }
    return try Set(String.fetchAll(db, sql: "SELECT identifier FROM grdb_migrations"))
}
```

Also note how the force unwrap of the Bool is just enough: SQLite will never return zero row or NULL.

Reply:
I've pushed up a new commit which does this.